### PR TITLE
pam_mktemp: init at 1.1.1

### PR DESCRIPTION
**Pull request by wladmis**
## Description of changes

pam_mktemp: init at 1.1.1

pam_mktemp is a PAM module that may be used with a PAM-aware login service to provide per-user private directories under /tmp as part of PAM session or account management. When an interactive (shell) session is started, a directory is created and the environment variables TMPDIR and TMP are set to the name of the directory. 

Homepage: https://www.openwall.com/pam/
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/pam_mktemp/default.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+  configurePhase = "";
+
+  buildInputs = [ pam e2fsprogs ];
+
+  makeFlags = [ "DESTDIR=$(out)" ];
+
+  enableParallelBuilding = true;
+
+  meta = with lib; {
```
Comment (Lurkki14) [inline pkgs/os-specific/linux/pam_mktemp/default.nix:41]: You need to add yourself as maintainer here

**Review by Lurkki14 — CHANGES_REQUESTED** [review]

File: pkgs/os-specific/linux/pam_mktemp/default.nix [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    (fetchpatch {
+      name = "allow_private_prefix_to_be_stricter.patch";
+      url = "https://git.altlinux.org/gears/p/pam_mktemp.git?p=pam_mktemp.git;a=commitdiff_plain;h=bb2cee0c695d22310e5364c30d74bccb0dbf3205";
+      hash = "sha256-TouysUVlNnl+m7lJ2VKPxUTYD2om1Jh5FEJ6NHMAI4U=";
+    })
+  ];
+
+  patchFlags = "-p2";
+
+  configurePhase = "";
```
Comment (h7x4) [inline pkgs/os-specific/linux/pam_mktemp/default.nix:33]: Please use `dontConfigure = true` if you need to disable the configure phase.

**Comment by Lurkki14** [comment]
The commit message for adding yourself to maintainers should be `maintainers: add <handle>` https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md

**Comment by wladmis** [comment]
> The commit message for adding yourself to maintainers should be `maintainers: add <handle>` https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md

Done.

File: pkgs/os-specific/linux/pam_mktemp/default.nix [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  dontConfigure = true;
+
+  buildInputs = [ pam e2fsprogs ];
+
+  makeFlags = [ "DESTDIR=$(out)" ];
+
+  enableParallelBuilding = true;
+
+  meta = with lib; {
+    homepage = "https://openwall.co/pam/";
```
Comment (Lurkki14) [inline pkgs/os-specific/linux/pam_mktemp/default.nix:42]: https://www.openwall.com/pam/ looks like the real URL

**Review by Lurkki14 — CHANGES_REQUESTED** [review]

File: pkgs/os-specific/linux/pam_mktemp/default.nix (hunk 3 above)
Comment (wladmis) [reply pkgs/os-specific/linux/pam_mktemp/default.nix:42]: Oops! Will fix it.

**Review by Lurkki14 — APPROVED** [review]

**Comment by wladmis** [comment]
Ping

**Review by cab404 — APPROVED** [review]

**Comment by cab404** [comment]
Thank you for the contribution! <3

**Pull request merged**